### PR TITLE
Get the Group Users for the Group Rules Engine

### PR DESCRIPTION
**Pull request by blueandgold**
Some notes:
(0) Only getting group members of USER type, which I think is relevant to the defined rule.

(1) It seems better to select from the GroupMember table, rather than the raw GroupMember json because the select statement can be (more easily) targeted toward fetching ACTIVE, USER only, vs fetching json.

(2) Not clear how to handle the hierarchy or parenting just yet.  So, leaving a TODO in place.  I'm sure this will become obvious in the next step (when doing traversals to find violations).

(3) Tried subclassing the group dao from the (base) dao.  It works, and the code is cleaner.  But there is still much more to be done here, or replaced by ORM.

(4) No tests here just yet, as this is still much in flux.  Will make sure there is test coverage before wrapping up work here.

### CODE REVIEW

File: google/cloud/security/common/gcp_type/group_member.py [hunk 1]
```diff
@@ -24,22 +24,23 @@
...
-        self.type = rule_def_member.get('type')
-        self.email = rule_def_member.get('email')
+        self.member_role = member_role
+        self.member_type = member_type
+        self.member_email = member_email
 
-        if (not self.role or
-                not self.type or
-                not self.email):
+        if (not self.member_role or
```
Comment (carise) [inline google/cloud/security/common/gcp_type/group_member.py]: nit: trailing whitespace

File: google/cloud/security/common/data_access/group_dao.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+from google.cloud.security.common.data_access import dao
+from google.cloud.security.common.data_access.sql_queries import select_data
+from google.cloud.security.common.gcp_type import group_member
+from google.cloud.security.common.util import log_util
+
+LOGGER = log_util.get_logger(__name__)
+
+
+class GroupDao(dao.Dao):
+    """Data access object (DAO) for Organizations."""
```
Comment (blueandgold) [inline google/cloud/security/common/data_access/group_dao.py]: Done.  Good catch.

File: google/cloud/security/common/gcp_type/group_member.py (hunk 1 above)
Comment (blueandgold) [inline google/cloud/security/common/gcp_type/group_member.py]: Done.

**Comment by blueandgold** [comment]
Adding `cla: yes` label, as the PR author is a googler at the time of this PR was created.

**Pull request merged**